proteus/BoundaryConditions.py

```diff
@@ -1,37 +1,42 @@
+import cython
 """
 Module for creating boundary conditions. Imported in SpatialTools.py
 
 .. inheritance-diagram:: proteus.BoundaryConditions
    :parts: 1
 """
 
-
-class BC_Base(object):
+class BC_Base():
     """
     Generic class regrouping boundary conditions
     """
-    def __init__(self, shape=None, name=None, b_or=None, b_i=None):
+    def __init__(self, shape=None, name=None, b_or=None, b_i=0, nd=None):
         self.Shape = shape
         self.name = name
         self.BC_type = 'None'
-        self._b_or = b_or  # array of orientation of all boundaries of shape
-        self._b_i = b_i  # indice for this boundary in list of boundaries
+        if shape is not None:
+          self.nd = self.Shape.Domain.nd
+        elif nd is not None:
+            self.nd = nd
+        else:
+            assert nd is not None, 'Shape or nd must be passed to BC'
+        if b_or is not None:
+            self._b_or = b_or[b_i]  # array of orientation of all boundaries of shape
 
-    @classmethod
-    def newGlobalBC(cls, name, default_value=None):
-        """
-        Makes a new BoundaryCondition class instance attached to BC_Base.
-        This creates a new class attributed that will be added to all BC_Base
-        class instances.
+    # @staticmethod
```

Comment:
I left it commented out there because these were useful member functions that could not work anymore once cythonised, but could potentially be revived if a solution is found.